hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3ABlockOutputStream.java

```diff
@@ -193,6 +198,7 @@ class S3ABlockOutputStream extends OutputStream implements
       initMultipartUpload();
     }
     this.isCSEEnabled = builder.isCSEEnabled;
+    this.ioStatisticsContext = builder.ioStatisticsContext;
```

Comment:
cache the thread iostats here and use in close(), something like `threadIOStatistics`